modules/codegen/src/main/scala/com/twilio/guardrail/generators/collections/JavaVavrCollectionsGenerator.scala

```diff
@@ -0,0 +1,85 @@
+import com.twilio.guardrail.terms.CollectionsLibTerms
+
+object JavaVavrCollectionsGenerator {
+  class JavaVavrCollectionsInterp extends CollectionsLibTerms[JavaLanguage, Target] {
+    private val baseInterp = new JavaCollectionsGenerator.JavaCollectionsInterp
+
+    override def MonadF: Monad[Target] = Target.targetInstances
+
+    override def vendorPrefixes(): Target[List[String]] = baseInterp.vendorPrefixes().map("x-java-vavr" +: _)
+
+    override def liftOptionalType(value: Type): Target[Type] =
+      safeParseClassOrInterfaceType("io.vavr.control.Option").map(_.setTypeArguments(new NodeList(value)))
+
+    override def liftOptionalTerm(value: Node): Target[Node] =
+      buildMethodCall("io.vavr.control.Option.of", Some(value))
+
+    override def liftSomeTerm(value: Node): Target[Node] =
+      for {
+        wrappedValue <- buildMethodCall("java.util.Objects.requireNonNull", Some(value))
+        term         <- buildMethodCall("io.vavr.control.Option.some", Some(wrappedValue))
+      } yield term
+
+    override def emptyOptionalTerm(): Target[Node] = buildMethodCall("io.vavr.control.Option.none")
+
+    override def arrayType(format: Option[String]): Target[Type] =
+      safeParseClassOrInterfaceType("io.vavr.collection.Vector").map(_.setTypeArguments(new NodeList[Type](STRING_TYPE)))
+
+    override def liftVectorType(value: Type, customTpe: Option[Type]): Target[Type] =
+      customTpe
+        .fold[Target[ClassOrInterfaceType]](safeParseClassOrInterfaceType("io.vavr.collection.Vector").map(identity))({
+          case t: ClassOrInterfaceType =>
+            Target.pure(t)
+          case x =>
+            Target.raiseUserError(s"Unsure how to map $x")
+        })
+        .map(_.setTypeArguments(new NodeList(value)))
+
+    override def liftVectorTerm(value: Node): Target[Node] =
+      buildMethodCall("io.vavr.collection.Vector.of", Some(value))
+
+    override def emptyArray(): Target[Node] = buildMethodCall("io.vavr.collection.Vector.empty")
+
+    override def embedArray(tpe: SwaggerUtil.LazyResolvedType[JavaLanguage], containerTpe: Option[Type]): Target[SwaggerUtil.LazyResolvedType[JavaLanguage]] =
+      tpe match {
+        case SwaggerUtil.Deferred(tpe) =>
+          Target.pure(SwaggerUtil.DeferredArray[JavaLanguage](tpe, containerTpe))
+        case SwaggerUtil.DeferredArray(_, _) =>
+          Target.raiseUserError("FIXME: Got an Array of Arrays, currently not supported")
+        case SwaggerUtil.DeferredMap(_, _) =>
+          Target.raiseUserError("FIXME: Got an Array of Maps, currently not supported")
+      }
+
+    override def liftMapType(value: Type, customTpe: Option[Type]): Target[Type] =
+      customTpe
+        .fold[Target[ClassOrInterfaceType]](safeParseClassOrInterfaceType("io.vavr.collection.Map").map(identity))({
+          case t: ClassOrInterfaceType =>
+            Target.pure(t)
+          case x =>
+            Target.raiseUserError(s"Unsure how to map $x")
```

Comment:
It'd be good to thread `Tracker` to this point (I suspect because interfaces with `SwaggerUtil` it'll be particularly unpleasant to do), but it would be a huge win for improving error messages.

Reply:
Yeah, looking at the call trees, there are 12 places that call this, and most of them are deep through `SwaggerUtil` :-/